### ISSUE TITLE
segfault in xsettings_notify_func 

### ISSUE THREAD

**Issue by solsticedhiver** [clutter-project/mx#41]
hi.
I have filled this bug https://bugzilla.gnome.org/show_bug.cgi?id=670842 against totem to be told it is a mx bug and come here.

here a copy of the stack trace

```
Program received signal SIGSEGV, Segmentation fault.
0xb73d3a8c in xsettings_notify_func (name=<optimized out>, action=XSETTINGS_ACTION_DELETED, setting=0x0, cb_data=0x8746cf0)
    at ../mx/x11/mx-settings-x11.c:299
299          if (g_strcmp0 (priv->font_name, setting->data.v_string))
#0  0xb73d3a8c in xsettings_notify_func (name=<optimized out>, action=XSETTINGS_ACTION_DELETED, setting=0x0, 
    cb_data=0x8746cf0) at ../mx/x11/mx-settings-x11.c:299
#1  0xb73d6066 in notify_changes (old_list=0x8888b30, client=0x8879a28) at ../mx/x11/xsettings-client.c:76
#2  read_settings (client=<optimized out>) at ../mx/x11/xsettings-client.c:404
#3  0xb73d6aed in xsettings_client_process_event (client=0x8879a28, xev=0xbfffee4c) at ../mx/x11/xsettings-client.c:597
#4  0xb73d3948 in mx_settings_x11_event_filter (xev=0xbfffee4c, cev=0x9133b80, data=0x8746cf0)
    at ../mx/x11/mx-settings-x11.c:489
#5  0xb6ef8658 in clutter_backend_x11_translate_event (backend=0x8197840, native=0xbfffee4c, event=0x9133b80)
    at ./x11/clutter-backend-x11.c:677
#6  0xb6f26fef in _clutter_backend_translate_event (backend=0x8197840, native=0xbfffee4c, event=0x9133b80)
    at ./clutter-backend.c:947
#7  0xb6efabc8 in clutter_x11_handle_event (xevent=0xbfffee4c) at ./x11/clutter-event-x11.c:316
#8  0xb6ff233b in gtk_clutter_filter_func (native_event=0xbfffee4c, event=0xb5603db8, user_data=0x8837058)
    at ./gtk-clutter-embed.c:190
#9  0xb7a43e83 in gdk_event_apply_filters (xevent=0xbfffee4c, event=0xb5603db8, window=0x0) at gdkeventsource.c:83
#10 0xb7a44032 in gdk_event_source_translate_event (xevent=0xbfffee4c, event_source=0x80a2de0) at gdkeventsource.c:197
#11 _gdk_x11_display_queue_events (display=0x809a000) at gdkeventsource.c:334
#12 0xb7a12e8e in gdk_display_get_event (display=0x809a000) at gdkdisplay.c:311
#13 0xb7a43cca in gdk_event_source_dispatch (source=0x80a2de0, callback=0, user_data=0x0) at gdkeventsource.c:356
#14 0xb7676c9e in g_main_dispatch (context=0x80a40d0) at gmain.c:2441
#15 g_main_context_dispatch (context=0x80a40d0) at gmain.c:3011
#16 0xb76773c0 in g_main_context_iterate (context=0x80a40d0, block=-1217897424, dispatch=1, self=<optimized out>)
    at gmain.c:3089
#17 0xb76779cb in g_main_loop_run (loop=0x81449f8) at gmain.c:3297
#18 0xb7bee045 in gtk_main () at gtkmain.c:1362
#19 0xb7aec2f4 in gtk_application_run_mainloop (application=0x8086690) at gtkapplication.c:115
#20 0xb794b2dc in g_application_run (application=0x8086690, argc=1, argv=0xbffff174) at gapplication.c:1323
#21 0x08049cbd in main (argc=1, argv=0xbffff174) at totem.c:280
```

**Comment by thos** [comment]
Thanks for reporting this issue. Could you run "bt full" to get a stack trace with all variables?

**Comment by ghost** [comment]
You're probably using Mx < 1.4.2, this appears to be fixed in 8f9ea55a4207934ac84deb7f2e98deb901f1cddd.

**Comment by solsticedhiver** [comment]
yes it seems to be fixed with 1.4.2
